diff --git a/meta/classes-recipe/image_types_wic.bbclass b/meta/classes-recipe/image_types_wic.bbclass
index cf3be909b30c..b7fcfc62605d 100644
--- a/meta/classes-recipe/image_types_wic.bbclass
+++ b/meta/classes-recipe/image_types_wic.bbclass
@@ -88,7 +88,16 @@ IMAGE_CMD:wic () {
 		esac
 		shift
 	done
-	mv "$build_wic/$(basename "${wks%.wks}")"*.${IMAGER} "$out.wic"
+	cp "$build_wic/$(basename "${wks%.wks}")"*.${IMAGER} "$out.wic"
+	wks_basename="$(basename "${wks%.wks}")"
+	cp "$build_wic/$wks_basename"*.direct "$out${IMAGE_NAME_SUFFIX}.wic"
+	for partition in "$build_wic/$wks_basename"*.direct.p2; do
+		partition_basename="$(basename "$partition")"
+		ext="${partition_basename##*.}"
+		gzip -f -9 -k "$partition"
+		partition_image_basename="${IMAGE_BASENAME}-${MACHINE}".direct."$ext"
+		cp "$partition".gz "${DEPLOY_DIR_IMAGE}/$partition_image_basename".gz
+	done
 }
 IMAGE_CMD:wic[vardepsexclude] = "WKS_FULL_PATH WKS_FILES TOPDIR"
 do_image_wic[cleandirs] = "${WORKDIR}/build-wic"
diff --git a/scripts/lib/wic/plugins/imager/direct.py b/scripts/lib/wic/plugins/imager/direct.py
index a1d152659b62..fbbea1299fc8 100644
--- a/scripts/lib/wic/plugins/imager/direct.py
+++ b/scripts/lib/wic/plugins/imager/direct.py
@@ -270,8 +270,7 @@ class DirectPlugin(ImagerPlugin):
                     return "/dev/%s%s%-d" % (part.disk, suffix, part.realnum)
 
     def cleanup(self):
-        if self._image:
-            self._image.cleanup()
+        logger.info("Intentionally disabled removing partition images.")
 
         # Move results to the output dir
         if not os.path.exists(self.outdir):
