From 01b03f8e909928fe0d3977e254e8d6032f8b9fc4 Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Fri, 15 Jul 2022 12:37:27 +0000
Subject: [PATCH] CMakeLists.txt: patch get_version.cmake and c_flags.cmake
 include path

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 064aa06..32dd809 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,11 +19,11 @@ set(PROJECT_DESCRIPTION "Open source version of the STMicroelectronics ST-LINK T
 include(GNUInstallDirs) # Define GNU standard installation directories
 
 ## Determine project version
-include(${CMAKE_MODULE_PATH}/get_version.cmake)
+include(${PROJECT_SOURCE_DIR}/cmake/modules/get_version.cmake)
 
 ## Set C build flags
 if (NOT MSVC)
-    include(${CMAKE_MODULE_PATH}/c_flags.cmake)
+	include(${PROJECT_SOURCE_DIR}/cmake/modules/c_flags.cmake)
 else ()
     message(STATUS "MSVC C Flags override to /MT")
     set(CMAKE_C_FLAGS_DEBUG_INIT          "/D_DEBUG /MTd /Zi /Ob0 /Od /RTC1")
