From dc90e35216fe2664fdffba924440579eadd1a4a2 Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Fri, 19 Jul 2024 10:53:04 +0000
Subject: [PATCH 7/9] drivers: spidev: add spidev to compatible list

---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index d13dc15cc..37e06975e 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -737,6 +737,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "lineartechnology,ltc2488", .data = &spidev_of_check },
 	{ .compatible = "lwn,bk4", .data = &spidev_of_check },
 	{ .compatible = "menlo,m53cpld", .data = &spidev_of_check },
+	{ .compatible = "spidev" },
 	{ .compatible = "micron,spi-authenta", .data = &spidev_of_check },
 	{ .compatible = "rohm,dh2228fv", .data = &spidev_of_check },
 	{ .compatible = "semtech,sx1301", .data = &spidev_of_check },
-- 
2.44.1

