From 9e83c1dfde8d9060b61ce9f8fa5c542671bf210e Mon Sep 17 00:00:00 2001
From: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>
Date: Wed, 4 Nov 2020 21:27:10 +0100
Subject: [PATCH 7/8] drivers: spidev: add spidev to compatible list

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>
---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index be503a0e6ef7..80c5c097b78f 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -678,6 +678,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "lwn,bk4" },
 	{ .compatible = "dh,dhcom-board" },
 	{ .compatible = "menlo,m53cpld" },
+	{ .compatible = "spidev" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.25.1

