--- a/pyproject.toml	2026-08-02 12:23:40.213767749 +0200
+++ b/pyproject.toml	2026-08-02 12:23:58.215738704 +0200
@@ -1,10 +1,8 @@
 [project]
 name = "annotated-doc"
 dynamic = []
-license = "MIT"
-license-files = [
-    "LICENSE",
-]
+license = { text = "MIT" }
+license-files = { paths = ["LICENSE"] }
 description = "Document parameters, class attributes, return types, and variables inline, with Annotated."
 readme = "README.md"
 authors = [
